packages/data-types/bin/data-types.js

```
@@ -0,0 +1,15 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const path = require('path');
+
+try {
+    // this path.join is only used for pkg asset injection
+    path.join(__dirname, '../package.json');
+    require('../dist/src/command');
+} catch (err) {
+    // eslint-disable-next-line
+    console.error('error while attempting to invoke cli command', err.toString());
```

This file has some odd behavior, first `path.join` will only throw an error if a non-string is passed and since the result is never being used, it is useless. Second I think try catching here is an odd choice, if there an expected error you should let it be thrown with a full stack track and by node, if you are trying to detect if the `../dist/src/command` exists, then you should use `fs.existsSync` first and throw an error that says the that library hasn't been built properly.

---

Most, if not all of the reasons it was done this way was to make this pkg(able). When we were working on ts-transforms we spent several tickets/issues to get this to work with the way pkg was being used. 'path.join' is only for 'pkg' to be able to notice that this is an additional asset to load in the environment. The error as well provided the error desired when dealing with pkg.